CMakeLists.txt

```diff
@@ -116,6 +117,34 @@ if( USE_OMP )
     endif()
 endif()
 
+#
+# Build Google Test
+#
+if( BUILD_GOOGLE_TEST )
+  # Control internal options of GoogleTest
+  #
+  set( INSTALL_GTEST OFF CACHE INTERNAL "Not install GoogleTest")
+  set( BUILD_GMOCK ON CACHE INTERNAL "Build gmock")
+
+  # Let's use the new mechanism to incorporate GoogleTest
+  #
+  include(FetchContent)
+  if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.24")
+    FetchContent_Declare( googletest
+      URL https://github.com/google/googletest/archive/refs/heads/main.zip
+      DOWNLOAD_EXTRACT_TIMESTAMP NEW )
+  endif()
```

Comment:
Hi @ifranda , could you remove the `if()` and `endif()` lines (line 132 and 136) for me? The CMake version should be specified at the beginning of the `CMakeLists.txt` (Line 7, `cmake_minimum_required()`) function, instead of here.

(I lost my access to update this PR.)